includes/commerce_pos.payment.inc

```diff
@@ -116,7 +128,7 @@ function commerce_pos_payment(&$form, &$form_state, $transaction_type) {
     );
 
     $form['edit_order'] = array(
-      '#markup' => l($return ? t('Edit Return') : t('Edit Order'), 'admin/commerce/pos/' . $form['#redirect_key'], array(
+      '#markup' => l($return ? t('Edit Return') : ($transaction_type == CommercePosService::TRANSACTION_TYPE_EXCHANGE ? t('Edit Exchange') : t('Edit Order')), 'admin/commerce/pos/' . $form['#redirect_key'], array(
```

Comment:
You could combine this logic to be something like this:

$transaction_types = CommercePosService::transactionTypes();

l(t('Edit @type', array($transaction_types[$transaction_type])), 'admin/commerce/pos/' . $transaction_type, array( ....